src/core/server/rendering/rendering_service.test.ts

```diff
@@ -0,0 +1,185 @@
+import { RenderingServiceSetup } from './types';
+import { RenderingService } from './rendering_service';
+
+const INJECTED_METADATA = {
+  version: expect.any(String),
+  branch: expect.any(String),
+  buildNumber: expect.any(Number),
+  env: {
+    binDir: expect.any(String),
+    configDir: expect.any(String),
+    homeDir: expect.any(String),
+    logDir: expect.any(String),
+    packageInfo: {
+      branch: expect.any(String),
+      buildNum: expect.any(Number),
+      buildSha: expect.any(String),
+      version: expect.any(String),
+    },
+    pluginSearchPaths: expect.any(Array),
+    staticFilesDir: expect.any(String),
+  },
+  legacyMetadata: {
+    branch: expect.any(String),
+    buildNum: expect.any(Number),
+    buildSha: expect.any(String),
+    version: expect.any(String),
+  },
+};
+const { createKibanaRequest, createRawRequest } = httpServerMock;
+const legacyApp = { getId: () => 'legacy' };
+
+describe('RenderingService', () => {
+  let service: RenderingService;
+
+  beforeEach(() => {
+    jest.clearAllMocks();
+    service = new RenderingService(mockRenderingServiceParams);
+  });
+
+  describe('setup()', () => {
+    it('creates instance of RenderingServiceSetup', async () => {
+      const rendering = await service.setup(mockRenderingSetupDeps);
+
+      expect(rendering.render).toBeInstanceOf(Function);
+    });
+
+    describe('render()', () => {
+      let uiSettings: ReturnType<typeof uiSettingsServiceMock.createClient>;
+      let render: RenderingServiceSetup['render'];
+
+      beforeEach(async () => {
+        uiSettings = uiSettingsServiceMock.createClient();
+        uiSettings.getRegistered.mockReturnValue({
+          registered: { name: 'title' },
+        });
+        render = (await service.setup(mockRenderingSetupDeps)).render;
+      });
+
+      it('renders "core" page', async () => {
+        const content = await render(createKibanaRequest(), uiSettings);
+        const dom = load(content);
+        const data = JSON.parse(dom('kbn-injected-metadata').attr('data'));
+
+        expect(data).toMatchSnapshot(INJECTED_METADATA);
+      });
+
+      it('renders "core" page for blank basepath', async () => {
+        mockRenderingSetupDeps.http.basePath.get.mockReturnValueOnce('');
+
+        const content = await render(createKibanaRequest(), uiSettings);
+        const dom = load(content);
+        const data = JSON.parse(dom('kbn-injected-metadata').attr('data'));
+
+        expect(data).toMatchSnapshot(INJECTED_METADATA);
+      });
+
+      it('renders "core" page driven by settings', async () => {
+        uiSettings.getUserProvided.mockResolvedValue({ 'theme:darkMode': { userValue: true } });
+        const content = await render(createKibanaRequest(), uiSettings);
+        const dom = load(content);
+        const data = JSON.parse(dom('kbn-injected-metadata').attr('data'));
+
+        expect(data).toMatchSnapshot(INJECTED_METADATA);
```

Comment:
The intention of the test is not clear. Let's check the data that necessary. Every update in injected metadata leads to false positive in the current state.

Reply:
This is by design, as we are encapsulating the intent in the test name and snapshot data.

https://elastic.slack.com/archives/C5TQ33ND8/p1576613947075400